src/mono/wasm/runtime/dotnet_crypto_worker.js

```diff
@@ -0,0 +1,173 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+var ChannelWorker = {
+    _impl: class {
+        // BEGIN ChannelOwner contract - shared constants.
+        get STATE_IDX() { return 0; }
+        get MSG_SIZE_IDX() { return 1; }
+
+        // Communication states.
+        get STATE_SHUTDOWN() { return -1; } // Shutdown
+        get STATE_IDLE() { return 0; }
+        get STATE_REQ() { return 1; }
+        get STATE_RESP() { return 2; }
+        get STATE_REQ_P() { return 3; } // Request has multiple parts
+        get STATE_RESP_P() { return 4; } // Response has multiple parts
+        get STATE_AWAIT() { return 5; } // Awaiting the next part
+        // END ChannelOwner contract - shared constants.
+
+        constructor(comm_buf, msg_buf, msg_char_len) {
+            this.comm = new Int32Array(comm_buf);
+            this.msg = new Uint16Array(msg_buf);
+            this.msg_char_len = msg_char_len;
+        }
+
+        async await_request(async_call) {
+            // console.log("await_request()");
+
+            for (;;) {
+                // Wait for signal to perform operation
+                Atomics.wait(this.comm, this.STATE_IDX, this.STATE_IDLE);
+
+                // Read in request
+                var req = this._read_request();
+                // console.log("Request: " + req);
+                if (req === this.STATE_SHUTDOWN)
+                    break;
```

Comment:
I don't know what style rules we have for JS code, but our C# guidelines here would require braces, since we're inside another braced statement (the for loop).

I think braces are a good idea.